tests/MongoDB.Driver.Tests/UpdateDefinitionBuilderTests.cs

```diff
@@ -265,6 +263,21 @@ public void Indexed_Positional_Typed()
             Assert(subject.Set(x => x.Pets.ElementAt(-1).Name, "Fluffencutters"), "{$set: {'pets.$.name': 'Fluffencutters'}}");
         }
 
+        [Fact]
+        public void Indexed_Typed()
+        {
+            var subject = CreateSubject<Person>();
+
+            Assert(subject.Set(x => x.FavoriteColors[2], "yellow"), "{$set: {'colors.2': 'yellow'}}");
+            Assert(subject.Set(x => x.Pets[2].Name, "Fluffencutters"), "{$set: {'pets.2.name': 'Fluffencutters'}}");
+            Assert(subject.Set(x => x.Pets.ElementAt(2).Name, "Fluffencutters"), "{$set: {'pets.2.name': 'Fluffencutters'}}");
+
+            var index = 2;
+            Assert(subject.Set(x => x.FavoriteColors[index], "yellow"), "{$set: {'colors.2': 'yellow'}}");
+            Assert(subject.Set(x => x.Pets[index].Name, "Fluffencutters"), "{$set: {'pets.2.name': 'Fluffencutters'}}");
+            Assert(subject.Set(x => x.Pets.ElementAt(index).Name, "Fluffencutters"), "{$set: {'pets.2.name': 'Fluffencutters'}}");
+        }
```

Comment:
This looks like a new test method, but isn't just an existing method moved to a new location?

Try to minimize diffs that result from simply moving things around, they make it harder to review.

Reply:
No, this is the old test that was just moved to another place to correspond to alphabetical order